segmentio/action-destinations
# fest(Segment): add all new endpoints to segment

We have added 3 new endpoints to talon-integration. This PR aims at adding these 3 endpoints to segment.

The new endpoints mainly support a new query param `skipNonExistingAttributes` compared to their previous versions.


This flag indicates whether to skip non-existing attributes.
If `true`, the non-existing attributes are skipped and a 400 error is not returned.
If `false`, a 400 error is returned in case of non-existing attributes.
## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
Please find the attached screenshots for the same. I have added screenshots of Action Tester with and without the `skipNonExistingAttributes` query param and the talon-ui after the API succeeds.
- Track Events Endpoint:
<img width="30%" alt="skipNonExistingAttributes=false" src="https://github.com/segmentio/action-destinations/assets/126659629/d6f01f0f-38c0-418b-9c3a-304ec48e98ba">
<img width="30%" alt="skipNonExistingAttributes=true" src="https://github.com/segmentio/action-destinations/assets/126659629/4628ba81-3a9c-42d1-8d5a-e89590224542">
<img width="30%" alt="talon-ui" src="https://github.com/segmentio/action-destinations/assets/126659629/7f909a5a-a680-479f-88ea-70290057b3ef">

- Update Customer Profile Endpoint:
<img width="24%" alt="skipNonExistingAttributes=false" src="https://github.com/segmentio/action-destinations/assets/126659629/fd2abf30-ced5-4605-b523-ed61bfa14e6f">
<img width="24%" alt="skipNonExistingAttributes=true" src="https://github.com/segmentio/action-destinations/assets/126659629/53f516c9-0c65-4fd9-8643-84346fffa464">
<img width="24%" alt="Screenshot 2023-08-16 at 16 19 37" src="https://github.com/segmentio/action-destinations/assets/126659629/4fd041fa-0272-4017-921f-94ffe79a3158">
<img width="24%" alt="Screenshot 2023-08-16 at 16 19 46" src="https://github.com/segmentio/action-destinations/assets/126659629/ff5fcd17-5a07-42bf-bc46-915bbb0fa8b1">

- Update Customer Session Endpoint:
<img width="30%" alt="skipNonExistingAttributes=false" src="https://github.com/segmentio/action-destinations/assets/126659629/aebe97d3-6610-4179-84df-9f499cdd7b10">
<img width="30%" alt="skipNonExistingAttributes=true" src="https://github.com/segmentio/action-destinations/assets/126659629/dff290f4-397c-4aee-bd6d-65104bd7e923">
<img width="30%" alt="Screenshot 2023-08-16 at 16 50 02" src="https://github.com/segmentio/action-destinations/assets/126659629/c90224bb-4bf1-4ed8-8195-db9ae6d04cf1">

- [ ] [Segmenters] Tested in the staging environment
